src/transformers/integrations/executorch.py

```diff
@@ -68,7 +68,7 @@ def __init__(self, model: PreTrainedModel):
             config=self.model.config,
             batch_size=self.model.generation_config.cache_config.batch_size,
             max_cache_len=self.model.generation_config.cache_config.max_cache_len,
-            dtype=self.model.config.torch_dtype,
+            dtype=self.model.dtype,
```

Comment:
I think this is more correct too -- model.config.torch_dtype specifies the dtype the weights were saved in, however all models will be loaded in fp32 by default. `model.dtype` reflects the actual dtype of the weights